lib/header.h

```diff
@@ -1069,7 +1069,7 @@
 #define PCI_DVSEC_ID_CXL	0	/* Designated Vendor-Specific ID for Intel CXL */
 
 /* PCIe CXL Designated Vendor-Specific Capabilities for Devices: Control, Status */
-#define PCI_CXL_DEV_LEN 0x38 /* CXL Device DVSEC Length */
+#define PCI_CXL_DEV_LEN 0x3c /* CXL Device DVSEC Length */
```

Comment:
This change looks suspicious. Either PCI_CXL_DEV_LEN is constant and should have its _constant_ value - which is not changing - and have its backward compatible ABI. Or it is variable (e.g. based on the current supported version in the lspci) and then it is not constant at all and should not be in header file defined as a compile time constant for all programs which use libpci API/ABI.

Reply:
The length is dependent on the revision, therefore isn't constant but always increasing.
Same reason, I followed the existing code from MR #120 that was accepted.

I considered `PCI_CXL_DEV_LEN` as the maximum length the current lspci version supports.
The actual length of the structure exposed by the device is read from the DVSEC header.

It's fine to keep it, it shows up to what length lspci supports the structure.